.github/workflows/test.yml

```diff
@@ -0,0 +1,123 @@
+        run: autopep8 --diff plugin/
+      - name: Install server dependencies
+        if: always()
+        run: cd server && yarn install
+      - name: Server lint check (ESlint)
+        if: always()
+        run: cd server && npx eslint --max-warnings=0 src/
+      - name: Server format check (Prettier)
+        if: always()
+        run: cd server && npx prettier --check 'src/**/*.{js,ts}'
+
+  script-method:
+    name: Test plugin (script method)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download yt-dlp
+        run: curl -L https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/latest/download/yt-dlp -o yt-dlp && chmod +x ./yt-dlp
+      - name: Install and build server dependencies
+        run: cd server/ && yarn install --frozen-lockfile && npx tsc
+      - name: Install plugin
+        run: |
+          mkdir ~/yt-dlp-plugins && curl -L "https://github.com/coletdjnz/yt-dlp-get-pot/releases/download/v0.1.1/yt-dlp-get-pot.zip" -o ~/yt-dlp-plugins/yt-dlp-get-pot.zip
+          cp -aT plugin/ ~/yt-dlp-plugins/bgutil-ytdlp-pot-provider/
+      - name: Test script method
+        shell: bash
+        run: |
+          script_response=$(./yt-dlp -vF --print-traffic --no-simulate -f "b*" -S "filesize:10M" --exec "rm -f" --extractor-args "youtube:getpot_bgutil_script=server/build/generate_once.js" dQw4w9WgXcQ 2>&1) || \
+            echo "::error title=Failed to run yt-dlp when testing script::\
+          yt-dlp returned $? exit status"
+
+          echo "::group::Logs from yt-dlp"
+          printf "%s\n" "$script_response"
+          echo "::endgroup::"
+          if [[ "$script_response" != *"BgUtilScriptPot: Generating POT via script: "* ]]; then
+            echo "::error title=BgUtilScriptPot was not invoked,file=plugin/yt_dlp_plugins/extractor/getpot_bgutil_script.py::\
+          BgUtilScriptPot was not invoked"
+            exit 1
+          fi
+
+  server-method:
+    name: Test plugin (server method)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download yt-dlp
+        run: curl -L https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/latest/download/yt-dlp -o yt-dlp && chmod +x ./yt-dlp
+      - name: Install plugin
+        run: |
+          mkdir ~/yt-dlp-plugins && curl -L "https://github.com/coletdjnz/yt-dlp-get-pot/releases/download/v0.1.1/yt-dlp-get-pot.zip" -o ~/yt-dlp-plugins/yt-dlp-get-pot.zip
+          cp -aT plugin/ ~/yt-dlp-plugins/bgutil-ytdlp-pot-provider/
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: server/
+          file: server/Dockerfile
+          push: false
+          tags: brainicism/bgutil-ytdlp-pot-provider:ci
+      - name: Test server method
+        timeout-minutes: 3
+        shell: bash
+        run: |
+          docker run --name bgutil-provider -d -p 4416:4416 brainicism/bgutil-ytdlp-pot-provider:ci
+          NODE_PID=$!
```

Comment:
NODE_PID is not used here, should we run docker in parallel and use docker stop instead?